src/transport.rs

```diff
@@ -148,23 +187,48 @@ impl Transport {
             _ => unreachable!(),
         }
     }
+
+    fn send_request(&self, req: Request<hyper::Body>) -> Result<hyper::Response<Body>> {
+        let req = match self {
+            Transport::Tcp { ref client, .. } => client.request(req),
+            Transport::EncryptedTcp { ref client, .. } => client.request(req),
+            Transport::Unix { ref client, .. } => client.request(req),
+        };
+
+        self.runtime().block_on(req).map_err(Error::Hyper)
+    }
+
+    fn runtime(&self) -> RefMut<Runtime> {
+        match self {
+            Transport::Tcp { ref runtime, .. } => runtime.borrow_mut(),
+            Transport::EncryptedTcp { ref runtime, .. } => runtime.borrow_mut(),
+            Transport::Unix { ref runtime, .. } => runtime.borrow_mut(),
+        }
+    }
 }
 
 /// Extract the error message content from an HTTP response that
 /// contains a Docker JSON error structure.
-fn get_error_message(res: &mut Response) -> Option<String> {
-    let mut output = String::new();
-    if res.read_to_string(&mut output).is_ok() {
-        let json_response = json::Json::from_str(output.as_str()).ok();
-        let message = json_response
-            .as_ref()
-            .and_then(|x| x.as_object())
-            .and_then(|x| x.get("message"))
-            .and_then(|x| x.as_string())
-            .map(|x| x.to_owned());
-
-        message
-    } else {
-        None
+fn get_error_message(res: Response<Body>) -> Option<String> {
+    let chunk = match res.into_body().concat2().wait() {
```

Comment:
Most of the other `wait` calls have been replaced with `self.runtime().block_on(req)` - should this one be too?

Reply:
hmm, I guess so?